### DISCUSSION TITLE
Codespaces suddenly not available via GitHub Classroom (Spending Limit) - One week left in semester

### DISCUSSION THREAD

**Discussion by aaryno** [community/community#40604] [GitHub Education]
I have been using GitHub Classroom for a couple years now and this year converted my class to use Codespaces. It has been going great but now that we have a week left, when I attempt to run a codespace for a new assignment I get this message: 
"You've used your included usage for this period. You can add a spending limit in [billing settings](https://github.com/settings/billing/spending_limit#codespaces-spending-limit)."

I've checked the org settings for Codespaces and this is set in settings:

Billing:
"All members and outside collaborators
Codespaces created for your organization's repositories by organization members and outside collaborators will be billed to the organization"

Policies:
Education Codespaces Benefit Policy - Machine Types: 2 cores

I can't seem to figure out what the org usage is or if/when it resets. Our usage has not been crazy. I have no idea how we are supposed to rely on this for a classroom if it's going to tap out with a week left in the semester. 

I have 5 days left in the semester and I can't make any new codespaces. I'm trying to record my last two lectures before the weekend. None of the students have reported having this issue yet. 

Reading up a little. In the GItHub Classroom docs for this class there is a reference to the Educator benefit policy ([link](https://docs.github.com/en/education/manage-coursework-with-github-classroom/integrate-github-classroom-with-an-ide/using-github-codespaces-with-github-classroom#about-the-codespaces-education-benefit-for-verified-teachers)):

"Note: The Codespaces Education benefit is currently in public beta and subject to change. During the beta release, your organization will not be charged if you exceed the free allowance."

"The Codespaces Education benefit gives verified teachers a free monthly allowance of GitHub Codespaces hours to use in GitHub Classroom. The free allowance is estimated to be enough for a class of 50 with 5 assignments per month, on a 2 core machine with 1 codespace stored per student."

I have 30 students and they are doing 2 assignments per week.

When I go to the organization's billing, the entry for Codespaces indicates 0 cores used this month and $0.00 charged. It's completely set to 0 for all machine types.
Anyway, this is kind of an urgent issue and I have no idea how to get help for this. I'm just an adjunct instructor teaching an open source computing class at a state university.

**Comment by spenserblack** [comment]
Since this sounds like it's an emergency for you, as a quick solution you can switch over to gitpod.io, which has a lot of similarities to GitHub Codespaces, until your billing / usage limit issue gets resolved.

**Comment by marklocklear** [comment]
Curious to hear if you. were able to resolve this @aaryno. I'm considering codespaces or my c++ classes. I am loving everything I am seeing but want to be sure it will be free for us.
